junit-platform-engine/src/main/java/org/junit/platform/engine/support/descriptor/FilePosition.java

```diff
@@ -52,6 +54,50 @@ public static FilePosition from(int line, int column) {
 		return new FilePosition(line, column);
 	}
 
+	/**
+	 * Create an optional {@code FilePosition} parsing the supplied
+	 * {@code query} string.
+	 *
+	 * <p>Examples of valid {@code query} strings:
+	 * <ul>
+	 *     <li>{@code "line=23"}</li>
+	 *     <li>{@code "line=23&column=42"}</li>
+	 * </ul>
+	 *
+	 * @param query the query string; may be {@code null}
+	 * @since 1.3
+	 */
+	public static Optional<FilePosition> fromQuery(String query) {
+		FilePosition result = null;
+		Integer line = null;
+		Integer column = null;
+		if (StringUtils.isNotBlank(query)) {
+			try {
+				for (String pair : query.split("&")) {
+					String[] data = pair.split("=");
+					if (data.length == 2) {
+						String key = data[0];
+						int value = Integer.valueOf(data[1]);
+						if (key.equals("line")) {
+							line = value;
+						}
+						if (key.equals("column")) {
+							column = value;
+						}
+					}
+				}
+			}
+			catch (IllegalArgumentException e) {
+				LoggerFactory.getLogger(FilePosition.class).debug(e, () -> "parsing query failed: " + query);
```

Comment:
"parsing query failed: " --> "Failed to parse 'line' and/or 'column' from query string: "

Reply:
Please store the logger in a static field.

Reply:
✔️